src/linking_number_certificate.cpp

```diff
@@ -363,9 +364,17 @@ void LinkingNumberCertificate::ComputeFromModel(const Model &model,
   std::vector<std::vector<int>> potential_links = PotentialLinkSearch(model);
   std::vector<Curve> discretized_curves;
   std::vector<std::vector<int>> potential_links_unique_per_curve;
+
+  if(model.get_segment_type() != SegmentType::Polyline){
```

Comment:
Please add a boolean argument to this function (e.g. `skip_discretization_for_polylines`) and check it here:
```
  if (!skip_discretization_for_polylines || (model.get_segment_type() != SegmentType::Polyline)) {
```
You can then add a flag in `define_args.h` and pass it in from `main.cpp`, to expose this as a command line argument.